<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Interactive And Collaborative Exploration Of Large-Scale Transcriptomics Data With </title>
    <meta charset="utf-8" />
    <meta name="author" content=" kevinrue  KevinRUE67  kevin-rue-albrecht" />
    <meta name="date" content="2021-11-05" />
    <script src="libs/header-attrs-2.11/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/metropolis.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/rladies-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
    <link href="libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
    <link rel="stylesheet" href="my-theme.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Interactive And Collaborative Exploration Of Large-Scale Transcriptomics Data With <a href="https://bioconductor.org/packages/iSEE/"><img src="img/isee-sticker.png" alt="logo" style="vertical-align:bottom" height="70px"></a>
## CCB talk 2021
### <i class="fab fa-github" style="color: #000000;"></i> <a href="https://kevinrue.github.io">kevinrue</a> <i class="fab fa-twitter" style="color: #1da1f2;"></i> <a href="https://twitter.com/KevinRUE67">KevinRUE67</a> <i class="fab fa-linkedin" style="color: #0e76a8;"></i> <a href="https://www.linkedin.com/in/kevin-rue-albrecht/">kevin-rue-albrecht</a>
### Kevin Rue-Albrecht, Sims Group
### 2021-11-05

---


layout: true

&lt;div class="my-header"&gt;&lt;a href="https://bioconductor.org/packages/iSEE/"&gt;&lt;img src="img/isee-sticker.png" alt="logo" align="right" height="100%"&gt;&lt;/a&gt;&lt;/div&gt;

&lt;div class="my-footer"&gt;&lt;span&gt;
Interactive And Collaborative Exploration Of Large-Scale Transcriptomics Data With iSEE
&lt;/span&gt;&lt;/div&gt;



---

# iSEE

![](img/pbmc3k-prettified.png)&lt;!-- --&gt;

---

# <svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:static;"><path d="M223.75 130.75L154.62 15.54A31.997 31.997 0 0 0 127.18 0H16.03C3.08 0-4.5 14.57 2.92 25.18l111.27 158.96c29.72-27.77 67.52-46.83 109.56-53.39zM495.97 0H384.82c-11.24 0-21.66 5.9-27.44 15.54l-69.13 115.21c42.04 6.56 79.84 25.62 109.56 53.38L509.08 25.18C516.5 14.57 508.92 0 495.97 0zM256 160c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm92.52 157.26l-37.93 36.96 8.97 52.22c1.6 9.36-8.26 16.51-16.65 12.09L256 393.88l-46.9 24.65c-8.4 4.45-18.25-2.74-16.65-12.09l8.97-52.22-37.93-36.96c-6.82-6.64-3.05-18.23 6.35-19.59l52.43-7.64 23.43-47.52c2.11-4.28 6.19-6.39 10.28-6.39 4.11 0 8.22 2.14 10.33 6.39l23.43 47.52 52.43 7.64c9.4 1.36 13.17 12.95 6.35 19.59z"/></svg> Winners of the 1st RStudio Shiny Contest

&lt;img src="img/shiny-contest.png" height="300px" style="display: block; margin: auto;" /&gt;

.center[
_"Most technically impressive"_
]

.footnote[
Read more: &lt;https://blog.rstudio.com/2019/04/05/first-shiny-contest-winners/&gt;
]

---

# &lt;i class="fas fa-users"&gt;&lt;/i&gt; Team effort

&lt;br/&gt;

&lt;table&gt;
&lt;tr&gt;
&lt;td width="25%"&gt;&lt;a href="https://kevinrue.github.io"&gt;&lt;img src="img/kevin-rue.jpeg"&gt;&lt;/a&gt;&lt;/td&gt;
&lt;td width="25%"&gt;&lt;a href="https://csoneson.github.io"&gt;&lt;img src="img/charlotte-soneson.png"&gt;&lt;/a&gt;&lt;/td&gt;
&lt;td width="25%"&gt;&lt;a href="https://federicomarini.github.io"&gt;&lt;img src="img/federico-marini.png"&gt;&lt;/a&gt;&lt;/td&gt;
&lt;td width="25%"&gt;&lt;a href="https://orcid.org/0000-0002-3564-4813"&gt;&lt;img src="img/aaron-lun.png"&gt;&lt;/a&gt;&lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td align='center' width="25%"&gt;&lt;a href="https://kevinrue.github.io"&gt;Kevin&lt;/a&gt;&lt;/td&gt;
&lt;td align='center' width="25%"&gt;&lt;a href="https://csoneson.github.io"&gt;Charlotte&lt;/a&gt;&lt;/td&gt;
&lt;td align='center' width="25%"&gt;&lt;a href="https://federicomarini.github.io"&gt;Federico&lt;/a&gt;&lt;/td&gt;
&lt;td align='center' width="25%"&gt;&lt;a href="https://orcid.org/0000-0002-3564-4813"&gt;Aaron&lt;/a&gt;&lt;/td&gt;
&lt;/tr&gt;
&lt;/table&gt;

.center[
&lt;a href="https://bioconductor.org/"&gt;
&lt;img src="img/bioconductor-sticker.png" alt="iSEE" height="200px"&gt;
&lt;/a&gt;
&lt;a href="https://bioconductor.github.io/EuroBioc2017/"&gt;
&lt;img src="img/EuroBioc2017-sticker.png" alt="iSEE" height="200px"&gt;
&lt;/a&gt;
]

---

# Bioconductor

.pull-left[
## Release 3.13 (April 2021)

- 2042 software packages
- 406 experiment data packages
- 965 annotation packages
- 29 workflows

&lt;https://bioconductor.org/&gt;


.center[
&lt;img src="img/bioconductor-sticker.png" alt="iSEE" height="200px"&gt;
]
]

.pull-right[
&lt;img src="img/osca.fig3.workflow.png" height="450px" /&gt;

.small-p[
[Orchestrating single-cell analysis with Bioconductor](https://doi.org/10.1038/s41592-019-0700-8)
&lt;a name=cite-amezquita2020osca&gt;&lt;/a&gt;([Amezquita, Lun, Becht, et al., 2020](https://www.ncbi.nlm.nih.gov/pubmed/31792435))
]
]

---

# SingleCellExperiment

![](img/osca.fig2.sce.png)&lt;!-- --&gt;

.small-p[
[Orchestrating single-cell analysis with Bioconductor](https://doi.org/10.1038/s41592-019-0700-8)
([Amezquita, Lun, Becht, et al., 2020](https://www.ncbi.nlm.nih.gov/pubmed/31792435))
]

---

# SingleCellExperiment &amp; iSEE

&lt;img src="img/osca-sce-isee.png" height="500px" style="display: block; margin: auto;" /&gt;

---

# Parameter boxes

![](img/pbmc3k-parametes-boxes.png)&lt;!-- --&gt;

---

# Transmitted selections 

![](img/pbmc3k-selection.png)&lt;!-- --&gt;

---

# Code tracker

![](img/pbmc3k-code-tracker.png)&lt;!-- --&gt;

---

# Panel settings

![](img/pbmc3k-panel-settings.png)&lt;!-- --&gt;

---

# Preconfigured applications

.pull-left[
## Define the initial state


```r
initial &lt;- list(
    ReducedDimensionPlot(
        PanelWidth = 6L,
        Type = "TSNE",
        ColorByColumnData = "Cluster"
    )
)
```
]

.pull-right[
## Alternatively

Use the panel settings exported from a past session.
]

## Configure the app and launch!


```r
app &lt;- iSEE(pbmc3k, initial = initial)
shiny::runApp(app)
```

---

# Guided tours

![](img/allen-tour.png)&lt;!-- --&gt;

---

# Developing new panels

![](img/panel-classes.png)&lt;!-- --&gt;

.right[
`AggregatedDotPlot`, `DynamicMarkerTable`, `DynamicReducedDimensionPlot`, ...
&lt;a href="https://bioconductor.org/packages/iSEEu/"&gt;
&lt;img src="img/iseeu-sticker.png" alt="logo" style="vertical-align:middle" height="80px"&gt;
&lt;/a&gt;
]

---

# Custom landing pages

![](img/landing-page.png)&lt;!-- --&gt;

---

# More

- Not only single-cell transcriptomics, but any rectangular experimental data set, e.g.
  
  + bulk assays (e.g., RNA-seq, ChIP-seq)

  + microarray
  
  + flow cytometry
  
  + ...

&lt;br/&gt;

- Voice recognition, with `iSEE(..., voice = TRUE)`

&lt;br/&gt;

- Online book [Extending iSEE](https://isee.github.io/iSEE-book/)

---

# References

.small-text[
&lt;a name=bib-amezquita2020osca&gt;&lt;/a&gt;[Amezquita, R. A., A. T. L. Lun, E.
Becht, et al.](#cite-amezquita2020osca) (2020). "Orchestrating
single-cell analysis with Bioconductor". In: _Nat Methods_ 17.2, pp.
137-145. ISSN: 1548-7105 (Electronic) 1548-7091 (Linking). DOI:
[10.1038/s41592-019-0654-x](https://doi.org/10.1038%2Fs41592-019-0654-x).
URL:
[https://www.ncbi.nlm.nih.gov/pubmed/31792435](https://www.ncbi.nlm.nih.gov/pubmed/31792435).

&lt;a name=bib-rue2018isee&gt;&lt;/a&gt;[Rue-Albrecht, K., F. Marini, C. Soneson,
et al.](#cite-rue2018isee) (2018). "iSEE: Interactive
SummarizedExperiment Explorer". In: _F1000Res_ 7, p. 741. ISSN:
2046-1402 (Print) 2046-1402 (Linking). DOI:
[10.12688/f1000research.14966.1](https://doi.org/10.12688%2Ff1000research.14966.1).
URL:
[https://www.ncbi.nlm.nih.gov/pubmed/30002819](https://www.ncbi.nlm.nih.gov/pubmed/30002819).
]

.center[
&lt;a href="https://bioconductor.org/packages/iSEE/"&gt;
&lt;img src="img/isee-sticker.png" alt="logo" style="vertical-align:middle" height="80px"&gt;
&lt;/a&gt;
]

---

# Other projects &amp; commitments

- [OBDS Training Programme](https://www.imm.ox.ac.uk/research/units-and-centres/mrc-wimm-centre-for-computational-biology/training/oxford-biomedical-data-science-training-programme)

- Software development &amp; maintenance

  + &lt;a href="https://github.com/kevinrue/velociraptor"&gt;
  &lt;img src="img/velociraptor-sticker.png" alt="logo" style="vertical-align:middle"   height="80px"&gt;
  &lt;/a&gt;
  &lt;a href="https://github.com/iSEE/iSEEu"&gt;
  &lt;img src="img/iseeu-sticker.png" alt="logo" style="vertical-align:middle"   height="80px"&gt;
  &lt;/a&gt;
  &lt;a href="https://github.com/iSEE/iSEE"&gt;
  &lt;img src="img/isee-sticker.png" alt="logo" style="vertical-align:middle"   height="80px"&gt;
  &lt;/a&gt;
  &lt;a href="https://github.com/kevinrue/TVTB"&gt;
  &lt;img src="img/tvtb-sticker.png" alt="logo" style="vertical-align:middle"   height="80px"&gt;
  &lt;/a&gt;
  &lt;a href="https://github.com/kevinrue/GOexpress"&gt;
  &lt;img src="img/goexpress-sticker.png" alt="logo" style="vertical-align:middle"   height="80px"&gt;
  &lt;/a&gt;

  + [CapCruncher](https://github.com/sims-lab/CapCruncher): Analysis software for Capture-C, Tri-C and Tiled-C data

- Member of the [Biocondutor Community Advisory Board](http://www.bioconductor.org/about/community-advisory-board/)

  + Multilingual Code of Conduct: &lt;https://github.com/kevinrue/bioc_coc_multilingual&gt;

  + Education Committe - The Carpentries lesson development: [The Bioconductor project](https://github.com/carpentries-incubator/bioc-project)

  + Online book: [Bioconductor Packages: Development, Maintenance, and Peer Review](http://contributions.bioconductor.org/)
  
  + Online materials: [Bioconductor Guidelines for Mentors and Mentees](https://kevinrue.github.io/bioc-mentorship-docs/)
  
- Conference talk: [NextGen Omics UK 2021](https://www.oxfordglobal.co.uk/nextgen-omics-series-uk/) (November 4-5 &amp; 8, 2021)
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
